File: sdk/dart/test/evaluation_api_test.dart
import 'package:built_collection/built_collection.dart';
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

import 'mock_dio_helper.dart';

/// tests for EvaluationApi
void main() {
  final client = createMockFlagentClient();
  final instance = client.getEvaluationApi();

  group(EvaluationApi, () {
    test('test postEvaluation', () async {
      final context = EvalContext((b) => b
        ..entityID = 'user-1'
        ..entityType = 'user'
        ..flagKey = 'test_flag');
      final response = await instance.postEvaluation(evalContext: context);
      expect(response.data, isNotNull);
      expect(response.data!.flagKey, equals('test_flag'));
      expect(response.data!.variantKey, equals('control'));
      expect(response.statusCode, equals(200));
    });

    test('test postEvaluationBatch', () async {
      final request = EvaluationBatchRequest((b) => b
        ..entities = ListBuilder<EvaluationEntity>([
          EvaluationEntity((e) => e..entityID = 'user-1'..entityType = 'user'),
        ])
        ..flagKeys = ListBuilder<String>(['test_flag']));
      final response = await instance.postEvaluationBatch(
        evaluationBatchRequest: request,
      );
      expect(response.data, isNotNull);
      expect(response.data!.evaluationResults, isEmpty);
      expect(response.statusCode, equals(200));
    });
  });
}

File: sdk/javascript/__tests__/EvaluationApi.test.ts
/**
 * Tests for EvaluationApi (generated OpenAPI client).
 */

import type { AxiosInstance } from 'axios';
import { Configuration } from '../configuration';
import { EvaluationApi } from '../api/evaluation-api';
import type { EvalContext, EvalResult } from '../models';

describe('EvaluationApi', () => {
  let mockAxios: jest.Mocked<AxiosInstance>;
  let api: EvaluationApi;

  beforeEach(() => {
    mockAxios = {
      request: jest.fn(),
      defaults: { baseURL: '' },
    } as unknown as jest.Mocked<AxiosInstance>;

    const config = new Configuration({
      basePath: 'https://api.example.com/api/v1',
    });

    api = new EvaluationApi(config, 'https://api.example.com/api/v1', mockAxios);
  });

  describe('postEvaluation', () => {
    it('should evaluate flag and return result', async () => {
      const context: EvalContext = {
        flagKey: 'test_flag',
        entityID: 'user1',
        entityType: 'user',
      };

      const expectedResult: EvalResult = {
        flagKey: 'test_flag',
        variantKey: 'control',
      };

      (mockAxios.request as jest.Mock).mockResolvedValue({
        data: expectedResult,
        status: 200,
      });

      const result = await api.postEvaluation(context);

      expect(result.data).toEqual(expectedResult);
      expect(mockAxios.request).toHaveBeenCalledTimes(1);
    });

    it('should pass correct request options', async () => {
      const context: EvalContext = { flagKey: 'f1' };
      (mockAxios.request as jest.Mock).mockResolvedValue({ data: {}, status: 200 });

      await api.postEvaluation(context);

      const call = (mockAxios.request as jest.Mock).mock.calls[0][0];
      expect(call.method?.toLowerCase()).toBe('post');
      expect(call.url).toContain('/evaluation');
      expect(call.data).toBeDefined();
    });
  });

  describe('postEvaluationBatch', () => {
    it('should evaluate batch and return results', async () => {
      const request = {
        entities: [{ entityID: 'user1' }],
        flagKeys: ['f1', 'f2'],
      };

      const expectedResponse = {
        evaluationResults: [
          { flagKey: 'f1', variantKey: 'control' },
          { flagKey: 'f2', variantKey: 'variant_a' },
        ],
      };

      (mockAxios.request as jest.Mock).mockResolvedValue({
        data: expectedResponse,
        status: 200,
      });

      const result = await api.postEvaluationBatch(request);

      expect(result.data.evaluationResults).toHaveLength(2);
      expect(result.data.evaluationResults[0].flagKey).toBe('f1');
    });
  });
});
